src/__tests__/requirement13.test.js

```diff
@@ -1,34 +1,32 @@
 import React from 'react';
 import { render, screen, waitFor, fireEvent } from '@testing-library/react';
 import App from '../App';
-import * as api from '../services/api';
-import mockedCategoriesResult from '../__mocks__/categories';
-import mockedQueryResult from '../__mocks__/query';
-
-jest.mock('../services/api');
-api.getCategories.mockImplementation(
-  () => Promise.resolve(mockedCategoriesResult),
-);
-api.getProductsFromCategoryAndQuery.mockImplementation(
-  () => Promise.resolve(mockedQueryResult),
-);
+import mockFetch from '../__mocks__/mockFetch';
 
 describe(`13 - Mostre junto ao ícone do carrinho a quantidade de produtos dentro dele, em todas as telas em que ele aparece`, () => {
   it('Vê a quantidade de produtos no carrinho da tela de listagem', async () => {
+
+    jest.spyOn(global, 'fetch').mockImplementation((url) => {
+      return mockFetch(url);
+    });
     render(<App />);
-    await waitFor(() => expect(api.getCategories).toHaveBeenCalled());
+    await waitFor(() => expect(global.fetch).toHaveBeenCalled());
     fireEvent.click(screen.getAllByTestId('category')[0]);
-    await waitFor(() => expect(api.getProductsFromCategoryAndQuery).toHaveBeenCalled());
+    await waitFor(() => expect(global.fetch).toHaveBeenCalled());
     fireEvent.click(screen.getAllByTestId('product-add-to-cart')[0]);
     fireEvent.click(screen.getAllByTestId('product-add-to-cart')[1]);
     expect(screen.getByTestId('shopping-cart-size')).toHaveTextContent('2');
   });
 
   it('Vê a quantidade de produtos no carrinho da tela de detalhes', async () => {
+
+    jest.spyOn(global, 'fetch').mockImplementation((url) => {
+      return mockFetch(url);
+    });
     render(<App />);
-    await waitFor(() => expect(api.getCategories).toHaveBeenCalled());
+    await waitFor(() => expect(global.fetch).toHaveBeenCalled());
     fireEvent.click(screen.getAllByTestId('category')[0]);
-    await waitFor(() => expect(api.getProductsFromCategoryAndQuery).toHaveBeenCalled());
+    await waitFor(() => expect(global.fetch).toHaveBeenCalled());
```

Comment:
:warning:  Aqui temos uma "pegadinha"

Como temos mais de um `it` dentro do mesmo `describe`, a gente precisa limpar o mock feito para que a contagem dê certo. 

Além do `toHaveBeenCalledTimes(2)` na linha 29, vamos precisar adicionar o seguinte trecho entre o `describe` (linha 6) e o `it` (linha 7):

```javascript
  afterEach(() => {
    global.fetch.mockClear();
  });
```
Dessa forma a gente limpa o mock feita depois de cada teste. Sem o `mockClear`, a contagem de chamadas do `fetch` nesse ponto seria 4 ao invés de 2.

```suggestion
    await waitFor(() => expect(global.fetch).toHaveBeenCalledTimes(2));
```